components/SearchBar/SearchBar.tsx

```diff
@@ -117,7 +120,11 @@ const SearchBar = ({
     }, 330);
 
   // Open Search, unless open then close/hide everything and remove explicit search width
+  const searchRef = createRef<HTMLInputElement>();
+
   const toggleSearch = () => {
+    const currSearchElement = searchRef.current!;
+    if (currSearchElement) { currSearchElement.focus() }
```

Comment:
`toggleSearch` "opens" or "closes" the search input, right?

If my assumption is right, and if I'd want to be nitpicking, shouldn't we then strictly speaking only need to set the focus when the search input is being opened, not when it's being closed (as in the latter case I assume you don't see the input anyway) ?

This question is more for my understanding than anything else, because of course it would be fine to always set the focus regardless whether we're "opening" or "closing".

Reply:
Yes line 27 is checking to see if `searchRef.current` still exists, when closing the ref gets destroyed and so condition gets bypassed and we do not set focus()